packages/ripple/index.js

```diff
@@ -30,19 +30,22 @@ const withRipple = (WrappedComponent) => {
       }
     }
 
-    initializeFoundation_ = (instance) => {
-      const adapter = this.createAdapter_(instance);
+    // surface: This element receives the visual treatment (classes and style) of the ripple.
+    // activator: This element is used to detect whether to activate the ripple. If this is not
+    // provided, the ripple surface will be used to detect activation.
+    initializeFoundation_ = (surface, activator) => {
+      const adapter = this.createAdapter_(surface, activator);
       this.foundation_ = new MDCRippleFoundation(adapter);
       this.foundation_.init();
     }
 
-    createAdapter_ = (instance) => {
+    createAdapter_ = (surface, activator) => {
       const MATCHES = util.getMatchesProperty(HTMLElement.prototype);
 
       return {
         browserSupportsCssVars: () => util.supportsCssVariables(window),
         isUnbounded: () => this.props.unbounded,
-        isSurfaceActive: () => instance[MATCHES](':active'),
+        isSurfaceActive: () => activator ? activator[MATCHES](':active') : surface[MATCHES](':active'),
```

Comment:
this is so simple and lovely. I wonder if we could simplify the Vanilla version of Checkbox and Switch in MDCW to look more like this. 
https://github.com/material-components/material-components-web/blob/master/packages/mdc-switch/index.js#L79
https://github.com/material-components/material-components-web/blob/master/packages/mdc-checkbox/index.js#L63

Reply:
We could probably modify `createAdapter` to take in an extra param as well